docs/lakefile.toml

```diff
@@ -0,0 +1,18 @@
+name = "batteries-docs"
+reservoir = false
+buildDir = "."
+defaultTargets = ["BatteriesDocs"]
+
+[[require]]
+name = "doc-gen4"
+git = "https://github.com/leanprover/doc-gen4"
+rev = "main"
+
+[[require]]
+name = "batteries"
+path = ".."
+
+[[lean_lib]]
+name = "BatteriesDocs"
```

Comment:
Why is `BatteriesDocs` in the root instead of in the `docs` folder?

Reply:
Yeah, this is a bit confusing. The `docs` folder is named this way just for easy integration with Github pages. It really should be the `BatteriesDocs` folder. Anyway, the reason is basically the same as why `Batteries.lean` is not in the `Batteries` folder.

Reply:
Also `lake clean` in `docs` deletes everything in `docs`, so it's better to have the minimum possible in there.

Reply:
Okay so maybe we should do that instead? Don't have a `docs/` folder at all in the repo, and treat it like a build directory. Put all the batteries docs stuff in a `BatteriesDocs` subfolder. I don't see the connection to batteries.lean being in the root, since the root of the `BatteriesDocs` project is the `BatteriesDocs/` or `docs/` folder.

Reply:
I don't understand how that's better?

Reply:
Wait! I think I have a bridge, give me a few minutes...

Reply:
Is this setup better?

Reply:
> I don't understand how that's better?

It means everything related to the `BatteriesDocs` project is in that folder, rather than living at the same root as `Batteries` (which implies that it is another Batteries lean_lib even though it isn't)

Reply:
> Is this setup better? (Do let me know since I will need to update CI and README accordingly. I can also revert if you don't think this is better.)

I still think the lean file should be in the folder. This is not standard lake layout, as indicated by the fact that you had to specify a `srcDir` for it. Normally the `srcDir` is `.` which is why the root file lives at the project root normally. In this package, the package root is `BatteriesDocs` so that's where the root file should be.

Reply:
> > I don't understand how that's better?
> 
> It means everything related to the `BatteriesDocs` project is in that folder, rather than living at the same root as `Batteries` (which implies that it is another Batteries lean_lib even though it isn't)

I don't think about it that way. The `docs` folder is for integration with github pages. Users of this scheme may want to put an entire website in there in addition to the doc-gen4 documentation. The `BatteriesDocs.lean` file is just a placeholder. The directory where the lakefile.toml is for BatteriesDocs is irrelevant: trying to build BatteriesDocs from any other directory will just fail. 

I can't think of a scenario where one would want to add dependencies to `BatteriesDocs.lean` other than importing the project roots, but the modification I just did allows for that peculiar scenario without interfering with website files.

Reply:
> > Is this setup better? (Do let me know since I will need to update CI and README accordingly. I can also revert if you don't think this is better.)
> 
> I still think the lean file should be in the folder. This is not standard lake layout, as indicated by the fact that you had to specify a `srcDir` for it. Normally the `srcDir` is `.` which is why the root file lives at the project root normally. In this package, the package root is `BatteriesDocs` so that's where the root file should be.

It's not a real package (hence `reservoir := false`). This will never be a "standard lake layout".

Reply:
I think we might be homing in on an issue with lake. While this is not the case right now, I can imagine `BatteriesTest.lean` also needing its own lakefile to support currently impossible tests. I think lake does currently allow specifying the config directory but I'm not sure how that works. I'll experiment with that over the next few days...

Reply:
> Maybe @tydeu can advise?

`lake -d <folder-with-lakefile> ...` will load the  workspace from the folder specified.

Reply:
Can we put that as a requirement in the base lakefile? In other words, have `BatteriesDocs` require a different lakefile to build itself?

Reply:
Perhaps `require "batteries-docs" from "BatteriesDocs"` is what you are looking for?

Reply:
No, that would reintroduce the circular dependencies.

Reply:
Oh, yeah. I am not sure what you want here, then. Either docs requires batteries or vice versa, and only one can be the root workspace  (at least at present).

Reply:
PS: This is a scapegoat for a general solution to the problem of generating docs for a project without users having to depend on doc-gen4. This (partial) solution is also good in the sense that doc-gen4 could depend on this project without a circular dependency issue.

Reply:
> I can't think of a scenario where one would want to add dependencies to `BatteriesDocs.lean` other than importing the project roots, but the modification I just did allows for that peculiar scenario without interfering with website files.

Let me put it this way: I want everything related to docs generation to be in one folder. I don't care if it's called `docs` or `BatteriesDocs` or something else, but I want it all neatly tucked away in one place where we can put a readme and so on. For reasons you have observed, `docs` is specifically problematic, so maybe not that. `BatteriesDocs` makes it look like a module directory, which it's not, so perhaps that's also a bad name and it should be `docgen` or `doc` or something else.

Reply:
Done. There's a chance it will look a bit weird at https://leanprover-community.github.io/batteries/doc but I guess we can punt that to doc-gen4.

Reply:
Henrik pointed out further simplifications that I think resolve all the issues discussed here.